BepInEx.Core/Contract/PluginInfo.cs

```diff
@@ -16,6 +16,11 @@ public class PluginInfo : ICacheable
     ///     General metadata about a plugin.
     /// </summary>
     public BepInPlugin Metadata { get; internal set; }
+    
+    /// <summary>
+    ///     The loader used to load this plugin, null if it is a provider
+    /// </summary>
+    public IPluginLoader Loader { get; internal set; }
```

Comment:
This could be null if the plugin was created later, e.g. by the ScriptEngine plugin whenever it gets ported to v6.